sympy/series/limits.py

```diff
@@ -326,13 +327,16 @@ def set_signs(expr):
         if z0 is S.Infinity:
             if e.is_Mul:
                 e = factor_terms(e)
-            newe = e.subs(z, 1/z)
+            dummy = Dummy('z', positive = z.is_positive, negative = z.is_negative, real = z.is_real)
+            newe = e.subs(z, 1/dummy)
             # cdir changes sign as oo- should become 0+
             cdir = -cdir
+            lead_term= dummy
```

Comment:
SymPy generally follows [PEP 8](https://peps.python.org/pep-0008/) which says to put a space before and after an equals sign in an assignment but no spaces around a keyword argument.